toml/src/main/kotlin/org/rust/toml/completion/LocalCargoTomlDependencyCompletionProvider.kt

```diff
@@ -6,25 +6,28 @@
 package org.rust.toml.completion
 
 import com.intellij.codeInsight.AutoPopupController
-import com.intellij.codeInsight.completion.CompletionResultSet
-import com.intellij.codeInsight.completion.CompletionSorter
-import com.intellij.codeInsight.completion.CompletionUtil
-import com.intellij.codeInsight.completion.PrioritizedLookupElement
+import com.intellij.codeInsight.completion.*
 import com.intellij.codeInsight.completion.impl.RealPrefixMatchingWeigher
 import com.intellij.codeInsight.lookup.LookupElement
 import com.intellij.codeInsight.lookup.LookupElementBuilder
 import com.intellij.codeInsight.lookup.LookupElementWeigher
 import com.intellij.icons.AllIcons
 import com.intellij.openapi.editor.EditorModificationUtil
+import com.intellij.util.ProcessingContext
 import org.rust.lang.core.completion.nextCharIs
+import org.rust.lang.core.psi.ext.ancestorStrict
 import org.rust.toml.StringValueInsertionHandler
+import org.rust.toml.crates.local.CargoRegistryCrateVersion
 import org.rust.toml.crates.local.CratesLocalIndexException
 import org.rust.toml.crates.local.CratesLocalIndexService
+import org.toml.lang.psi.TomlKeySegment
 import org.toml.lang.psi.TomlKeyValue
+import org.toml.lang.psi.TomlTable
 
 class LocalCargoTomlDependencyCompletionProvider : TomlKeyValueCompletionProviderBase() {
     override fun completeKey(keyValue: TomlKeyValue, result: CompletionResultSet) {
-        val prefix = CompletionUtil.getOriginalElement(keyValue.key)?.text ?: return
+        val keySegment = keyValue.key.segments.singleOrNull() ?: return
+        val prefix = CompletionUtil.getOriginalElement(keySegment)?.text ?: return
```

Comment:
Could you please add a test for these change? Currently, when I change these lines back, all tests pass

Reply:
It prevents possible completion for complex segments like `cratename.something.else = "<caret>"`. Because `cratename.version = "<caret>"` is a valid dependency declaration, but it doesn't seem to be popular, for now I think we may just ignore it